docs/index.md

```diff
@@ -0,0 +1,476 @@
+
+### Uploading directly from the 'My Content' page
+
+The second method for getting a locally-saved mobile map package to ArcGIS Enterprise portal or on ArcGIS Online is to 'pull' it using the [Add Item](http://doc.arcgis.com/en/arcgis-online/share-maps/add-items.htm) tool. This can be found from the 'My Content' page and is as simple as browsing for the local file, providing a title and some tags, and clicking 'ADD ITEM'.
+
+# App developer patterns
+
+Now that the mobile map package has been created and published, it can be downloaded by the app using an authenticated connection.
+
+## Authentication
+
+When the mapbook app launches, it checks for the mobile map package (.mmpk) on the device.  The app will attempt to download the mobile map package from Portal. Two different options are available in this app for authentication: Integrated Windows Authentication (IWA) and OAuth2. Choosing the authentication type is done by setting the option in the applications's Settings file. It is then handled in a switch inside the `DownloadViewModel`.
+
+```csharp
+switch (Settings.Default.AuthenticationType)
+{
+    // For Portal with IWA authentication enabled
+    case "IWA":
+        ...
+        break;
+
+    // For Portal with OAuth authentication enabled
+    case "OAuth":
+        ...
+        break;
+
+    // For Portal with no authentication enabled.
+    case "None":
+    default:
+        break;
+}
+```
+
+### IWA
+
+When [Portal is set up using IWA](http://server.arcgis.com/en/portal/latest/administer/windows/use-integrated-windows-authentication-with-your-portal.htm), users do not have to sign in to the app, they are automatically signed in using the same accounts they used to sign in to Windows.
+The app grabs the authenticated Windows credential and passes it to the ArcGIS Authentication Manager together with the Portal URI to authenticate the user
+
+```csharp
+// Get default credential for the authenticated Windows user
+var networkCredential = CredentialCache.DefaultCredentials.GetCredential(serviceUri, "Basic");
+
+// Create ArcGIS Network credential
+ArcGISNetworkCredential arcGisCredential = new ArcGISNetworkCredential
+{
+     Credentials = networkCredential,
+     ServiceUri = serviceUri,
+};
+
+// Add the credential to the AuthenticationManager
+Esri.ArcGISRuntime.Security.AuthenticationManager.Current.AddCredential(arcGisCredential);
+```
+
+### OAuth 2.0
+
+The default option for the app is OAuth 2.0 authentication. The app prompts the user for their organization’s ArcGIS Online credentials used to obtain a token later consumed by the Portal. The [ArcGIS Runtime API documentation](https://www.developers.arcgis.com/net/latest/wpf/guide/use-the-authentication-manager.htm) provides a detailed explanation on how to use the Authentication Manager to handle authentication in the app.
```

Comment:
> https://www.developers.arcgis.com/net/latest/wpf/guide/use-the-authentication-manager.htm

This URL results in a 404 when clicked. Can we remove the `www.`?
https://developers.arcgis.com/net/latest/wpf/guide/use-the-authentication-manager.htm